### PR TITLE
build: more compilation flags cleanup

### PR DESCRIPTION
**Pull request by nerzhul**
* Replaced remaining #ifndef trhing #define thing #end
* Drop 2 irrlicht internal compilation flags which are enabled unconditionally
* remove some useless #IRR_COMPILE checks on android

### CODE REVIEW

**Comment by nerzhul** [comment]
note: current failing jobs are already failing on master branch
See: https://github.com/minetest/irrlicht/actions/runs/6411693370/job/17407654243

File: include/Keycodes.h [hunk 1]
```diff
@@ -90,7 +89,7 @@ namespace irr
 		KEY_KEY_X            = 0x58,  // X key
 		KEY_KEY_Y            = 0x59,  // Y key
 		KEY_KEY_Z            = 0x5A,  // Z key
-		KEY_LWIN             = 0x5B,  // Left Windows key (Microsoft� Natural� keyboard)
+		KEY_LWIN             = 0x5B,  // Left Windows key (Microsoft� Natural� keyboard)
```
Comment (numberZero) [inline include/Keycodes.h:92]: mojibake

**Review by numberZero — APPROVED** [review]
LGTM otherwise

**Comment by numberZero** [comment]
> note: current failing jobs are already failing on master branch

This is what #250 is ought to fix

**Comment by nerzhul** [comment]
rebased

**Comment by sfan5** [comment]
Merged w/ squash

**Pull request closed without merging**